cloudsync/sync/manager.py

```diff
@@ -200,7 +200,19 @@ def _sync_one_entry(self, sync: SyncEntry):
             self.backoff()  # raises a backoff error to the caller
         return something_got_done
 
+    def _validate_provider_roots(self):
+        try:
+            for p in [LOCAL, REMOTE]:
+                if not self.providers[p].root_validated:
```

Comment:
probably we don't want to support the storage this state flag inside the provider.   we should have our own root_validated[] state.